flaml/model.py

```diff
@@ -753,22 +748,109 @@ def _init_model_for_predict(self, X_test):
         return new_trainer, test_dataset, training_args
 
     def predict_proba(self, X, **kwargs):
+        from datasets import Dataset
+
         self._update_hf_args(kwargs)
         assert (
             self._task in CLASSIFICATION
         ), "predict_proba() only for classification tasks."
 
-        new_trainer, test_dataset, _ = self._init_model_for_predict(X)
+        X_test, _ = self._preprocess(X, **self._kwargs)
+        test_dataset = Dataset.from_pandas(X_test)
+
+        new_trainer, test_dataset, _ = self._init_model_for_predict(test_dataset)
         predictions = new_trainer.predict(test_dataset)
         return predictions.predictions
 
+    def evaluate(self, X_val: DataFrame, y_val: Series, **kwargs):

```

Comment:
Is `score` a better name for this method? Shall we make the signature compatible with the `score()` In scikit-learn? See #466 .